packages/tangle/ledgerstate.go

```diff
@@ -266,6 +269,26 @@ func (l *LedgerState) Consumers(outputID ledgerstate.OutputID) (cachedTransactio
 	return l.UTXODAG.CachedConsumers(outputID)
 }
 
+// ConfirmedConsumer returns the (cached) confirmed consumer of the given outputID.
+func (l *LedgerState) ConfirmedConsumer(outputID ledgerstate.OutputID) (confirmedConsumer *ledgerstate.CachedConsumer) {
+	found := false
+	for _, cachedConsumer := range l.Consumers(outputID) {
+		cachedConsumer.Consume(func(consumer *ledgerstate.Consumer) {
+			// We already identified a confirmed Consumer
+			l.CachedTransactionMetadata(consumer.TransactionID()).Consume(func(transactionMetadata *ledgerstate.TransactionMetadata) {
+				if l.tangle.ConfirmationOracle.IsBranchConfirmed(transactionMetadata.BranchID()) {
+					found = true
+					confirmedConsumer = cachedConsumer
```

Comment:
You already consumed `cachedConsumer` and therefore it might be not in cache anymore when it is returned. I'd suggest changing the method to return a `transactionID`.

Reply:
I have to have a serious look on how this ObjectStorage thing works!